### PR TITLE
Types for gulp-mustache

### PR DESCRIPTION
**Pull request by CAD97**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by CAD97** [comment]
Fixed the test file which had a wrong name.

**Comment by sheetalkamat** [comment]
@CAD97 The failure is in the tests file:

> 874Error: 'gulp-mustache/gulp-mustache-tests.ts' unnecessarily references 'node', which is already referenced in the type definition.
> 
> 875    at Object.<anonymous> (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/module-info.ts:380:11)
> 
> 876    at Generator.next (<anonymous>)
> 
> 877    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/module-info.js:4:58)
> 
> 878    at <anonymous>
>

**Comment by CAD97** [comment]
I can `squash` if desired. Fix was just removing extraneous `/// <reference types="node" />`.

File: types/gulp-mustache/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Transform } from "stream";
+
+declare namespace GulpMustache {
+    type View = { [key: string]: string } | string | undefined;
+
+    interface Partials {
```
Comment (sheetalkamat) [inline types/gulp-mustache/index.d.ts]: Might want to rename this as Hash instead? and write View as:
```TypeScript
type View = Hash | string | undefined;
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by sheetalkamat** [comment]
@CAD97 Please do not squash the changes. We will do that while merging.

File: types/gulp-mustache/index.d.ts (hunk 1 above)
Comment (CAD97) [inline types/gulp-mustache/index.d.ts]: 👍

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**